### DISCUSSION TITLE
How to include just one file from a subdirectory

### DISCUSSION THREAD

**Discussion by Evanshion** [marcelotduarte/cx_Freeze#1778] [Q&A]
Hi all, I have this structure:
Main application

- SAM_SOCKET (that I have to add with all subfolders)
    - Subfolder1
    - subfolder2
    - ...
    - windows/packets/X (that I have to add and contains several file but I need just one)

I did my setup this way, how to get just the interested file keeping the structure?

```python
import sys
import os
import cx_Freeze 



executables = [cx_Freeze.Executable("voicechecker.py", icon="Cefla.ico", target_name="VoiceAssistant.exe")]

include_files = ["img/", "Languages/", "Reports/", "SAM_SOCKET/", "Voices/", "Cefla.ico", "config.ini", "VoiceChecker.bat",
    ("SAM_SOCKET/windows/packets/9766LCEN_LangCore_windows_1_0_0.dat", "SAM_SOCKET/windows/packets/9766LCEN_LangCore_windows_1_0_0.dat")
    ]

build_exe_options = {"excludes": ["PyQt5", "PyQt6", "PyInstaller", "scipy", "SAM_SOCKET/windows/packets"],
                     "optimize": 1,
                     "include_files": include_files
                     }

cx_Freeze.setup(
    name = "Cefla Voice Assistant",
    options = {"build_exe": build_exe_options},
    author = "Giuseppe Ieno: giuseppe.ieno.ext@cefla.it",
    version = "1.0",
    description = "Tool for automated testing for Cefla vocal commands (Vivoka)",
    executables = executables
)
```

**Comment by marcelotduarte** [comment]
>     ("SAM_SOCKET/windows/packets/9766LCEN_LangCore_windows_1_0_0.dat", "SAM_SOCKET/windows/packets/9766LCEN_LangCore_windows_1_0_0.dat")

This line is correct, but a previous inclusion of "SAM_SOCKET/" includes all files below.
The `excludes` option is incorrect, it excludes modules, so, if "SAM_SOCKET/windows/packets" is a module or package, it should be "SAM_SOCKET.windows.packets"

**Reply by Evanshion** [reply]
mhh...I tried as you suggested, but I continue to see, in the subfolder /SAM_SOCKET/windows/packets a lot of files and not just the one I want...
My idea was to add manually the file I want and exclude then the folder, what I'm doing bad?

`executables = [cx_Freeze.Executable("voicechecker.py", icon="Cefla.ico", target_name="VoiceAssistant.exe")]

include_files = ["img/", "Languages/", "Reports/", "SAM_SOCKET/", "Voices/", "Cefla.ico", "config.ini", "VoiceChecker.bat",
    ("SAM_SOCKET/windows/packets/9766LCEN_LangCore_windows_1_0_0.dat", "SAM_SOCKET/windows/packets/9766LCEN_LangCore_windows_1_0_0.dat")
    ]

build_exe_options = {"excludes": ["PyQt5", "PyQt6", "PyInstaller", "scipy", "SAM_SOCKET.windows.packets"],
                     "optimize": 1,
                     "include_files": include_files
                     }

cx_Freeze.setup(
    name = "Cefla Voice Assistant",
    options = {"build_exe": build_exe_options},
    author = "Giuseppe Ieno: giuseppe.ieno.ext@cefla.it",
    version = "1.0",
    description = "Tool for automated testing for Cefla vocal commands (Vivoka)",
    executables = executables
)
`

**Reply by marcelotduarte** [reply]
Observe "previous inclusion of "SAM_SOCKET/" includes all files below."
And **if** "SAM_SOCKET/windows/packets" is a module

Probably you should use "SAM_SOCKET/Subfolder1", "SAM_SOCKET/Subfolder2" and so on.

**Reply by Evanshion** [reply]
mh...I suspected so...I thought there could be a more fast way!
Thank you so much!
